### PR TITLE
[CRA + RXJS + SC]: Create repo folder navigation

### PR DESCRIPTION
**Pull request by iansamz**
Resolves: #328

### CODE REVIEW

File: cra-rxjs-styled-components/src/components/file-explorer/FileExplorer.tsx [hunk 1]
```diff
@@ -13,14 +13,34 @@ import {
...
+import { FileItem } from '../../types/types';
 
 type Props = {
   branch: string;
-  dirNames: string[];
-  fileNames: string[];
+  path: string;
+  basePath: string;
+  directories: Array<FileItem>;
+  files: Array<FileItem>;
```
Comment (vyktoremario) [inline cra-rxjs-styled-components/src/components/file-explorer/FileExplorer.tsx:23]: Why not `FileItem[]`?

Comment (iansamz) [reply cra-rxjs-styled-components/src/components/file-explorer/FileExplorer.tsx]: Sorry was Copilot 😆

**Review by vyktoremario — APPROVED** [review]
LGTM. But is this also suppose to cover `FileViewer`? Like clicking on a file and seeing the content and the path becomes, `/blob/*`? 
if not, you can merge this.

**Comment by iansamz** [comment]
> LGTM. But is this also suppose to cover `FileViewer`? Like clicking on a file and seeing the content and the path becomes, `/blob/*`? if not, you can merge this.

Its fine @dustinsgoodman said we should separate them into. two prs. Let me update the pr info

**Pull request merged**